src/wallet/coinselection.cpp

```diff
@@ -367,13 +379,16 @@ util::Result<SelectionResult> KnapsackSolver(std::vector<OutputGroup>& groups, c
         }
 
         // If the result exceeds the maximum allowed size, return closest UTXO above the target
-        if (result.GetWeight() > max_weight) {
+        if (result.GetWeight() > max_input_weight) {
             // No coin above target, nothing to do.
             if (!lowest_larger) return ErrorMaxWeightExceeded();
 
             // Return closest UTXO above target
             result.Clear();
             result.AddInput(*lowest_larger);
+
+            // We may still fail if max is low
+            if (result.GetWeight() > max_input_weight) return ErrorMaxWeightExceeded();
```

Comment:
I don’t think this condition can be hit. 

AFAICT, there is only be a single scenario in which `vfBest` can contain an input selection that is in excess of the `max_input_weight`: `ApproximateBestSubset` initializes `vfBest` with all of `applicable_groups` without checking the weight. If it then never finds a better input selection throughout the exploration, it would be possible that it returns an overweight `vfBest`. Since `lowest_larger` is only set when it adheres to the weight limit, we either have replaced `vfBest` with an adequate `lowest_larger` at this point, or have already returned `ErrorMaxWeightExceeded`.

Reply:
makes sense, removed